src/Symfony/Component/Messenger/Event/WorkerMessageFailedEvent.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Event;
+
+use Symfony\Component\Messenger\Envelope;
+
+/**
+ * @experimental in 4.3
+ */
+class WorkerMessageFailedEvent extends AbstractWorkerMessageEvent
+{
+    private $throwable;
+
+    private $wasRequeued;
+
+    public function __construct(Envelope $envelope, string $receiverName, \Throwable $error, bool $wasRequeued)
+    {
+        $this->throwable = $error;
+        $this->wasRequeued = $wasRequeued;
+
+        parent::__construct($envelope, $receiverName);
+    }
+
+    public function getThrowable(): \Throwable
+    {
+        return $this->throwable;
+    }
+
+    public function wasRequeued(): bool
```

Comment:
Needs to be named willRequeue as the event is currently dispatched before the requeue actually happens